framework/src/media/audio/audio_manager.c

```diff
@@ -1140,14 +1259,22 @@ audio_manager_result_t reset_audio_stream_in(void)
 	card = &g_audio_in_cards[g_actual_audio_in_card_id];
 	if ((card->config[card->device_id].status == AUDIO_CARD_IDLE) || (card->config[card->device_id].status == AUDIO_CARD_NONE)) {
 		meddbg("Card status is wrong status : %d\n", card->config[card->device_id].status);
-		return AUDIO_MANAGER_INVALID_DEVICE;	
+		return AUDIO_MANAGER_INVALID_DEVICE;
 	}
 	pthread_mutex_lock(&(card->card_mutex));
 
+	device_channel_num = pcm_get_channels(card->pcm);
```

Comment:
both recorder & play change state to ready when it stopped.
It means it will not call prepare(set_stream_in/out).
So error will occur

Reply:
So, If it is stopped, memset is enough. free should be done in unprepare.
Same mechanism needed for resampling.

Reply:
Hmm.. Why the error will occur since pcm_get_channels() is called before pcm_close() is called..?